pkg/operator/highcpuusagesnoalertcontroller/highcpuusagesnoalert_controller.go

```diff
@@ -0,0 +1,104 @@
+package highcpuusagesnoalertcontroller
+
+import (
+	"bytes"
+	"context"
+	"strconv"
+	"time"
+
+	v1 "github.com/openshift/api/config/v1"
+	monitoringv1 "github.com/openshift/api/monitoring/v1"
+	configv1informers "github.com/openshift/client-go/config/informers/externalversions/config/v1"
+	configlistersv1 "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/cluster-kube-apiserver-operator/bindata"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/dynamic/dynamicinformer"
+)
+
+type highCPUUsageSNOController struct {
+	client               dynamic.Interface
+	infraLister          configlistersv1.InfrastructureLister
+	clusterVersionLister configlistersv1.ClusterVersionLister
+}
+
+func NewHighCPUUsageSNOAlertController(
+	configInformer configv1informers.Interface,
+	dynamicInformer dynamicinformer.DynamicSharedInformerFactory,
+	client dynamic.Interface,
+	recorder events.Recorder,
+) factory.Controller {
+	c := &highCPUUsageSNOController{
+		client:               client,
+		infraLister:          configInformer.Infrastructures().Lister(),
+		clusterVersionLister: configInformer.ClusterVersions().Lister(),
+	}
+
+	prometheusAlertInformer := dynamicInformer.ForResource(monitoringv1.SchemeGroupVersion.WithResource("prometheusrules"))
```

Comment:
Is this the correct GVR ? I don't think so.

I have checked the operator logs from a CI run and it thinks the same:
```
W0604 13:14:07.950220       1 reflector.go:547] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: failed to list monitoring.openshift.io/v1, Resource=prometheusrules: the server could not find the requested resource
E0604 13:14:07.950260       1 reflector.go:150] k8s.io/client-go@v0.30.0/tools/cache/reflector.go:232: Failed to watch monitoring.openshift.io/v1, Resource=prometheusrules: failed to list monitoring.openshift.io/v1, Resource=prometheusrules: the server could not find the requested resource
```